cylc/flow/scheduler.py

```diff
@@ -1623,6 +1623,7 @@ async def main_loop(self) -> None:
             self.broadcast_mgr.expire_broadcast(self.pool.get_min_point())
             self.late_tasks_check()
 
+            self.process_workflow_db_queue()
```

Comment:
Why this extra call?

Reply:
Need to ensure we've processed anything pending in the workflow DB queue otherwise this logic might not work if two messages are close together.